src/backend/distributed/shared_library_init.c

```diff
@@ -2722,4 +2724,17 @@ CitusObjectAccessHook(ObjectAccessType access, Oid classId, Oid objectId, int su
 		 * regardless if it's citus being created */
 		SetCreateCitusTransactionLevel(GetCurrentTransactionNestLevel());
 	}
+	else if (access == OAT_DROP && classId == ConstraintRelationId &&
+			 ConstraintWithIdIsOfType(objectId, CONSTRAINT_FOREIGN))
+	{
+		Oid relationId = ConstraintOnRelationId(objectId);
+		if (OidIsValid(relationId) && IsCitusTable(relationId))
```

Comment:
since this is a top-level entry point, do we need a CitusHasBeenLoaded somewhere?